### ISSUE TITLE
MaterialX: Sdr can not provide node information for ND_convert_float_color3

### ISSUE THREAD

**Issue by JGamache-autodesk** [PixarAnimationStudios/USD#1629]
### Description of Issue

The usdMtlx Sdr parser goes thru one level of indirection to get the mx::NodeDef by asking the mx::NodeGraph or mx::Implementation for a mx::NodeDef, and sometimes the indirect call fails.

### Steps to Reproduce
1. With a MaterialX enabled build of USD, open UsdView and do the following in the interpreter window:
```
>>> from pxr import Sdr
>>> r = Sdr.Registry()
>>> [i for i in r.GetNodeNames() if not r.GetNodeByName(i)]
['LightFilter', 'PluginLightFilter', 'ND_disney_brdf_2012_surface', 'ND_disney_bsdf_2015_surface', 'ND_add_displacementshader', 'ND_add_volumeshader', 'ND_arrayappend_integer_integerarray', 'ND_arrayappend_integerarray_integerarray', 'ND_arrayappend_float_floatarray', 'ND_arrayappend_floatarray_floatarray', 'ND_arrayappend_color3_color3array', 'ND_arrayappend_color3array_color3array', 'ND_arrayappend_color4_color4array', 'ND_arrayappend_color4array_color4array', 'ND_arrayappend_vector2_vector2array', 'ND_arrayappend_vector2array_vector2array', 'ND_arrayappend_vector3_vector3array', 'ND_arrayappend_vector3array_vector3array', 'ND_arrayappend_vector4_vector4array', 'ND_arrayappend_vector4array_vector4array', 'ND_arrayappend_string_stringarray', 'ND_arrayappend_stringarray_stringarray', 'ND_curveadjust_float', 'ND_curveadjust_color3', 'ND_curveadjust_color4', 'ND_curveadjust_vector2', 'ND_curveadjust_vector3', 'ND_curveadjust_vector4', 'ND_mix_displacementshader', 'ND_mix_volumeshader', 'ND_convert_float_color3', 'ND_convert_float_color4', 'ND_convert_float_vector2', 'ND_convert_float_vector3', 'ND_convert_float_vector4', 'ND_convert_vector3_color3', 'ND_convert_vector4_color4', 'ND_convert_color3_vector3', 'ND_convert_color4_vector4', 'ND_convert_color3_color4', 'ND_convert_color4_color3', '11753386459202641452']
```

All the buggy MaterialX nodes have in common that the NodeGraph/Implementation they use is missing a link back to the NodeDef they represent.

### To fix:

Go directly from Document to NodeDef in UsdMtlxParserPlugin::Parse(const NdrNodeDiscoveryResult&) using:
```
    auto nodeDef = document->getNodeDef(discoveryResult.identifier.GetString());
    if (!nodeDef) {
        TF_WARN("Invalid MaterialX NodeDef; unknown node name ' %s '",
            discoveryResult.identifier.GetText());
        return GetInvalidNode(discoveryResult);
    }

    ShaderBuilder builder(discoveryResult);
    ParseElement(&builder, nodeDef);
```
and remove the two ParseElement functions that uses `const mx::ConstNodeGraphPtr&` and `const mx::ConstImplementationPtr& impl` in parser.cpp.

**Comment by spiffmon** [comment]
Thanks for reporting, @JGamache-autodesk  - would you like to submit a PR for the fix?

**Comment by JGamache-autodesk** [comment]
Hi @spiffmon, I will start the process to get a PR submitted. First one from me, so will require some time to get all the details right.

**Comment by spiffmon** [comment]
No worries - we'll wait for it, and thank you!

**Comment by jilliene** [comment]
Filed as internal issue #USD-6913

**Comment by pmolodo** [comment]
I assume this issue can be closed now?

**Comment by spiffmon** [comment]
Indeed - the fix made it into 22.03 . Sorry, our automation will close the PR but not, I think the Issue, when there are both.
